From 30babd40047546a3c45991078eccb6b6d37bb130 Mon Sep 17 00:00:00 2001
From: Alessandro Astone <ales.astone@gmail.com>
Date: Sun, 4 Oct 2020 21:59:52 +0200
Subject: [PATCH 09/17] recovery: Don't consider manual install aborts as
 errors

 * From android-11, recovery warns the user about the danger of
   rebooting after a failed update, but if we didn't even start
   the install procedure it doesn't really make much sense.

Change-Id: Ifaf5eae9fe6e1d915a035e2570c22260de59f99b
---
 install/adb_install.cpp  | 2 +-
 install/fuse_install.cpp | 4 ++--
 recovery.cpp             | 3 +++
 3 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/install/adb_install.cpp b/install/adb_install.cpp
index b12e52944..5948e7958 100644
--- a/install/adb_install.cpp
+++ b/install/adb_install.cpp
@@ -347,7 +347,7 @@ InstallResult ApplyFromAdb(Device* device, bool rescue_mode, Device::BuiltinActi
 
   RecoveryUI* ui = device->GetUI();
 
-  InstallResult install_result = INSTALL_ERROR;
+  InstallResult install_result = INSTALL_NONE;
   std::map<MinadbdCommand, CommandFunction> command_map{
     { MinadbdCommand::kInstall, std::bind(&AdbInstallPackageHandler, device, &install_result) },
     { MinadbdCommand::kRebootAndroid, std::bind(&AdbRebootHandler, MinadbdCommand::kRebootAndroid,
diff --git a/install/fuse_install.cpp b/install/fuse_install.cpp
index 72c11ce98..06948998d 100644
--- a/install/fuse_install.cpp
+++ b/install/fuse_install.cpp
@@ -295,14 +295,14 @@ InstallResult ApplyFromSdcard(Device* device) {
 
   if (do_sdcard_mount() != 0) {
     LOG(ERROR) << "\nFailed to mount sdcard\n";
-    return INSTALL_ERROR;
+    return INSTALL_NONE;
   }
 
   std::string path = BrowseDirectory(SDCARD_ROOT, device, ui);
   if (path.empty()) {
     LOG(ERROR) << "\n-- No package file selected.\n";
     ensure_path_unmounted(SDCARD_ROOT);
-    return INSTALL_ERROR;
+    return INSTALL_NONE;
   }
 
   // Hint the install function to read from a block map file.
diff --git a/recovery.cpp b/recovery.cpp
index 789e3c8ec..3f81ccc80 100644
--- a/recovery.cpp
+++ b/recovery.cpp
@@ -577,6 +577,9 @@ static Device::BuiltinAction PromptAndWait(Device* device, InstallResult status)
         if (status == INSTALL_REBOOT) {
           return reboot_action;
         }
+        if (status == INSTALL_NONE) {
+          update_in_progress = false;
+        }
 
         if (status == INSTALL_SUCCESS) {
           update_in_progress = false;
-- 
2.47.1

